### PR TITLE
docs: add LeaderWorkerSet API reference

### LINKED ISSUE #169: Create API reference doc
**What would you like to be added**:

See https://github.com/kubernetes-sigs/jobset/pull/611 as an example how this was generated for JobSet.

For now, we can add it in a directory named `reference` under https://github.com/kubernetes-sigs/lws/tree/main/docs

**Why is this needed**:

So that we can easily reference LWS API fields.

### PR DESCRIPTION
**Pull request by zmvictor**
#### What this PR does / why we need it

Create API reference doc to easily reference LWS API fields

#### Which issue(s) this PR fixes

Fixes #169 

#### Special notes for your reviewer

```
make generate-apiref
```

#### Does this PR introduce a user-facing change?

NONE

### CODE REVIEW

File: hack/genref/markdown/pkg.tpl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+{{- range $idx, $val := .packages -}}
+{{/* Special handling for config */}}
+  {{- if .IsMain -}}
+---
+title: {{ .Title }}
+content_type: tool-reference
+package: {{ .DisplayName }}
+auto_generated: true
+description: Generated API reference documentation for {{ if ne .GroupName "" -}} {{ .DisplayName }}{{ else -}} Kueue Configuration{{- end -}}.
```
Comment (googs1025) [inline hack/genref/markdown/pkg.tpl:11]: this part `Kueue Configuration` We shouldn't need

Comment (zmvictor) [reply hack/genref/markdown/pkg.tpl]: Updated. PTAL

Comment (googs1025) [reply hack/genref/markdown/pkg.tpl]: do we have `LeaderWorkerSet Configuration` ? if not, we should remove this part

Comment (zmvictor) [reply hack/genref/markdown/pkg.tpl]: see https://github.com/kubernetes-sigs/lws/issues/170

**Review by kerthcet — COMMENTED** [review]
/approve

please squash the commits.

Leave LGTM to @liurupeng

**Comment by liurupeng** [comment]
Was out of office last week, thanks @zmvictor and @kerthcet !

**Pull request merged**